sdk/appconfiguration/azure-appconfiguration-provider/azure/appconfiguration/provider/_azureappconfigurationprovider.py

```diff
@@ -261,25 +273,222 @@ def _is_json_content_type(content_type: str) -> bool:
+        self._max_backoff: int = kwargs.pop("max_backoff", 600)
+
+    def reset(self) -> None:
+        self._next_refresh_time = time.time() + self._interval
+        self._attempts = 1
+
+    def retry(self) -> None:
+        self._next_refresh_time = time.time() + self._calculate_backoff()/1000
+        self._attempts += 1
+
+    def needs_refresh(self) -> bool:
+        return time.time() >= self._next_refresh_time
+
+    def _calculate_backoff(self) -> float:
+        max_attempts = 30
+        millisecond = 1000  # 1 Second in milliseconds
+
+        min_backoff_milliseconds = self._min_backoff * millisecond
+        max_backoff_milliseconds = self._max_backoff * millisecond
+
+        if self._attempts <= 1 or self._max_backoff <= self._min_backoff:
+            return min_backoff_milliseconds
+
+        calculated_milliseconds = max(1, min_backoff_milliseconds) * (1 << min(self._attempts, max_attempts))
+
+        if calculated_milliseconds > max_backoff_milliseconds or calculated_milliseconds <= 0:
+            calculated_milliseconds = max_backoff_milliseconds
+
+        return min_backoff_milliseconds + (
+            random.uniform(0.0, 1.0) * (calculated_milliseconds - min_backoff_milliseconds)
+        )
+
+
 class AzureAppConfigurationProvider(Mapping[str, Union[str, JSON]]):
     """
     Provides a dictionary-like interface to Azure App Configuration settings. Enables loading of sets of configuration
     settings from Azure App Configuration into a Python application. Enables trimming of prefixes from configuration
     keys. Enables resolution of Key Vault references in configuration settings.
     """
 
-    def __init__(self) -> None:
+    def __init__(self, **kwargs) -> None:
         self._dict: Dict[str, str] = {}
         self._trim_prefixes: List[str] = []
         self._client: Optional[AzureAppConfigurationClient] = None
         self._secret_clients: Dict[str, SecretClient] = {}
+        self._key_vault_options: Optional[AzureAppConfigurationKeyVaultOptions] = kwargs.pop("key_vault_options", None)
+        self._selects: List[SettingSelector] = kwargs.pop(
+            "selects", [SettingSelector(key_filter="*", label_filter=EMPTY_LABEL)]
+        )
+
+        trim_prefixes: List[str] = kwargs.pop("trim_prefixes", [])
+        self._trim_prefixes = sorted(trim_prefixes, key=len, reverse=True)
+
+        refresh_all: List[Tuple[str, str]] = kwargs.pop("refresh_all", None) or []
+        refresh_only: List[Tuple[str, str]] = kwargs.pop("refresh_only", None) or []
+        self._refresh_all: Mapping[Tuple[str, str]: Optional[str]] = {_build_sentinel(s): None for s in refresh_all}
+        self._refresh_only: Mapping[Tuple[str, str]: Optional[str]] = {_build_sentinel(s): None for s in refresh_only}
+        self._refresh_timer: _RefreshTimer = _RefreshTimer(**kwargs)
+        self._on_refresh_error: Optional[Callable[[Exception], None]] = kwargs.pop("on_refresh_error", None)
```

Comment:
We should also add this as a keyword-arg in the `load` method. Sorry I forgot to add that.